src/app.ts

```diff
@@ -183,9 +187,47 @@ export async function newApp(
   registerAPIRoutes(app, APIBasePath, allowNotifyIPSourceRange);
   registerPagoPARoutes(app, PagoPABasePath, allowPagoPAIPSourceRange);
 
+  const timer = idpMetadataUpdater(app);
+  app.on("server:stop", () => {
+    clearInterval(timer);
+  });
   return app;
 }
 
+export function idpMetadataUpdater(
+  app: Express,
+  listener?: () => void
```

Comment:
let's call this something more meaningful like `onRefresh`

Reply:
fixed in 82e8752